### ISSUE TITLE
Pipeline StableDiffusionSAGPipeline is not compatible with most of schedulers

### ISSUE THREAD

**Issue by vladmandic** [huggingface/diffusers#6443]
### Describe the bug

Like title says, pipeline `StableDiffusionSAGPipeline` is not compatible with most of samplers - basically only `PNDM`, `DDPM` and `DDIM` work, all others produce the error.

Given this is a built-in pipeline, not a community one I'd expect anything but base samplers to work.

### Reproduction

![xyz_grid-xyz_grid-0018-rcnz-cartoon-3d-v2-](https://github.com/huggingface/diffusers/assets/57876960/41f047d4-6e9f-4561-a6c8-087fb4ca6c25)

### Logs

```shell
Traceback (most recent call last)

│ /home/vlado/dev/sdnext-dev/modules/processing_diffusers.py:467 in process_diffusers
│   466 │   │   t0 = time.time()
│ ❱ 467 │   │   output = shared.sd_model(**base_args) # pylint: disable=not-callable
│   468 │   │   downcast_openvino(op="base")
│                                                             
│ /home/vlado/.local/lib/python3.11/site-packages/torch/utils/_contextlib.py:115 in decorate_context
│   114 │   │   with ctx_factory():
│ ❱ 115 │   │   │   return func(*args, **kwargs)
│   116                                                       
│ /home/vlado/.local/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion_sag/pipeline_stable_diffusion_sag.py:744 in __call__
│   743 │   │   │   │   │   │   │   # DDIM-like prediction of x0      
│ ❱ 744 │   │   │   │   │   │   │   pred_x0 = self.pred_x0(latents, noise_pred_uncond, t)
│   745 │   │   │   │   │   │   │   # get the stored attention maps
│ /home/vlado/.local/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion_sag/pipeline_stable_diffusion_sag.py:840 in pred_x0
│   839 │   def pred_x0(self, sample, model_output, timestep):
│ ❱ 840 │   │   alpha_prod_t = self.scheduler.alphas_cumprod[timestep]
│   841

RuntimeError: indices should be either on cpu or on the same device as the indexed tensor (cpu)
```

### System Info

diffusers==0.25.0

### Who can help?

@yiyixuxu @DN6 @sayakpaul @patrickvonplaten

**Comment by sayakpaul** [comment]
I agree. But at the same time, its usage is quite low based on our stats. Do you see a lot of use for it?

**Comment by vladmandic** [comment]
i do see quite a lot of asks for `sag_scale` in general. i don't think it needs to be a separate pipeline (and that's why usage is most likely very low), ideally it should be a param in standard pipeline.

**Comment by patrickvonplaten** [comment]
Hey @vladmandic, 

Thanks for the error message! Note that the way the SAG pipeline is written, schedulers of type "Karras et al." won't work. However, there is no reason why the other schedulers don't work. This is fixed here: #6465.

All DPM, UniPC, DEIS, ... - type schedulers are now supported.

**Comment by patrickvonplaten** [comment]
Also we throw a better error message now for Karras-style schedulers
